sunpy/__init__.py

```diff
@@ -45,7 +45,7 @@ class UnsupportedPythonError(Exception):
     from sunpy.util import system_info
     from sunpy.tests.runner import SunPyTestRunner
 
-    self_test = SunPyTestRunner.make_test_runner_in(os.path.dirname(__file__))
+    self_test = SunPyTestRunner.make_test_runner_in(str(Path(__file__).parent))
```

Comment:
I don't like to have to `str` Paths... does astropy's `make_test_runner` understand paths?

Reply:
Umm no it didnt. Initially i tried doing it without str but it threw errors. I have minimized str as much as possible.

Reply:
IF errors are thrown without a `str`, we need to know where and why so we can fix it at a later date.

Reply:
it was in all the functions, they didnt accept Path objects as input

Reply:
What functions errored?

Reply:
wherever i have used str, i tried earlier without it and if it gave errors, then i used str